Add and update doc.go files throughout kubelet

tcharding opened the pull request:
**What this PR does / why we need it**:

As noted in issue #11663 kubelet currently does not contain package doc comments for all packages.

This PR adds doc comments to all top level packages within kubelet; either as a `doc.go` file or, for packages with a single source file within that source file. Also, cleans up current doc strings to be consistent across the kubelet package, including some grammatical fixes.

**Special notes for your reviewer**:

I am new to the k8s project, please review wording to ensure it is correct. Comments are at times brief since I am not totally across the code base.

tcharding committed:
Add and update doc.go files throughout kubelet

Signed-off-by: tcharding <me@tobin.cc>

euank commented:
/unassign

tcharding commented:
/sig node
/kind documentation

tcharding commented:
/assign @Random-Liu

tcharding commented:
/release-note-none

resouer commented:
/ok-to-test

resouer commented:
/ok-to-test

tcharding commented:
/test pull-kubernetes-verify

tcharding commented:
rebased

tcharding commented:
I have no idea what to do about this failing test `pull-kubernetes-verify`. I updated each BUILD file and added an entry in `src` slice for `"doc.go"`. I have confirmed this by running

`find . -type d -exec doc.sh {} \;`

where `doc.sh` is

```bash
#!/bin/bash
#
# Verify "doc.go" is present in each BUILD file

cwd=$(pwd)

cd $1

cat doc.go &> /dev/null

if [[ $? == 0 ]]
then
    grep 'doc\.go' BUILD > /dev/null
    if [[ $? == 1 ]]
    then
        echo "BUILD entry missing for $PWD"
    fi
fi

cd $cwd
```

Any suggestions on where to dig?

euank commented:
@tcharding The output can be a bit tricky to parse, but it's trying to tell you to run `./hack/update-bazel.sh`. That should fix it.

tcharding commented:
thanks @euank, will proceed as suggested.

tcharding commented:
rebased and ran `hack/update-bazel.sh`. Added BUILD changes as additional commit.

Pushed with `git push -f`; Is this correct procedure?, `push` after rebase fails. I have been advised on this already but failed to understand the correct procedure for pushing to a branch after rebase.

euank commented:
Yes, force pushing after rebase is how you update github PRs @tcharding.

I'd probably also squash together the two most recent commits (which will also require another force push).

tcharding commented:
cool, thanks @euank

andyxning commented:
/test pull-kubernetes-e2e-kops-aws

andyxning commented on pkg/kubelet/volumemanager/doc.go:
```diff
@@ -0,0 +1,19 @@
+/*
+Copyright 2016 The Kubernetes Authors.
```
@tcharding Should this be updated to `2017`?

tcharding replied:
Good catch

tcharding replied:
There are a bunch of these errors. 2015, 2016 etc. Will change all to 2017 and re-submit

tcharding committed:
Update bazel
